components/service/glean/src/main/java/mozilla/components/service/glean/debug/GleanDebugActivity.kt

```diff
@@ -33,16 +43,37 @@ class GleanDebugActivity : Activity() {
 
         // Enable debugging options and start the application.
         intent.extras?.let {
+            // Check for ping debug view tag to apply to the X-Debug-ID header when uploading the
+            // ping to the endpoint
+            var pingTag: String? = intent.getStringExtra(TAG_DEBUG_VIEW_EXTRA_KEY)
+            var altUrl = "https://incoming.telemetry.mozilla.org"
```

Comment:
This should go right next to where we set `pingTag = null`. Otherwise, if the application is using a non-default server, it will reset it to the value of `altUrl` to *our* default every time the debug activity is called, even with options other than `tagPings`.

To recap, I think that this should init as `var altUrl: String? = null` and be reset to `altUrl = Configuration().serverEndpoint` after we set `pingTag = null`, below.

Reply:
I'll still have to reset the serverEndpoint every time, even when there is no tagPings flag...  Otherwise:
1. User tags ping (server endpoint changed to alt endpoint)
2. User sends new intent with no `tagPings` flag (server endpoint is still the same alt endpoint)

If I don't explicitly set it to the default each time then it never gets reset.

Reply:
That all makes sense -- but I wonder if there's a way to define this constant in one place.  I'm worried that this won't necessarily stay in sync with the one in `Configuration.kt`.

Reply:
@mdboom I think that using `Configuration().serverEndpoint` works around this, but using some constant is another good take on this!

Reply:
Indeed, I shall find a nice const home for it :)

Reply:
This has pretty much been refactored out of this class to the HttpPingUploader class, but I did think that having endpoints defined as constants was a good idea so I added a companion object with `DEFAULT_AWS_ENDPOINT` and `DEFAULT_GCP_ENDPOINT` constants